types/src/block_executor/partitioner.rs

```diff
@@ -9,16 +9,18 @@ pub type ShardId = usize;
 pub type TxnIndex = usize;
 
 #[derive(Debug, Clone, Eq, Hash, PartialEq, Serialize, Deserialize)]
-pub struct TxnIdxWithShardId {
+pub struct TxnIndexForSharding {
     pub txn_index: TxnIndex,
     pub shard_id: ShardId,
+    pub round: usize,
```

Comment:
nit: May be add an alias type for round as well.

Reply:
fixed